python/cuml/test/explainer/test_gpu_treeshap.py

```diff
@@ -493,3 +508,178 @@ def test_lightgbm_classifier_with_categorical(n_classes):
+                                              'reg:pseudohubererror']))
+            model = xgb.XGBRegressor(
+                n_estimators=n_estimators, tree_method='gpu_hist',
+                objective=objective, enable_categorical=True, verbosity=0).fit(
+                X, y)
+        elif task == 'classification':
+            valid_objectives = ['binary:logistic', 'binary:hinge',
+                                'binary:logitraw', 'count:poisson', ]
+            if n_targets > 2:
+                valid_objectives += ['rank:pairwise', 'rank:ndcg', 'rank:map',
+                                     'multi:softmax', 'multi:softprob']
+
+            objective = draw(st.sampled_from(valid_objectives))
+            model = xgb.XGBClassifier(
+                n_estimators=n_estimators, tree_method='gpu_hist',
+                objective=objective, enable_categorical=True, verbosity=0).fit(
+                X, y)
+        return model.get_booster(), model.predict(X, output_margin=True)
+    elif learner == 'rf':
+        predict_model = 'GPU 'if y.dtype == np.float32 else 'CPU'
+        if task == 'regression':
+            model = cuml.ensemble.RandomForestRegressor(
+                n_estimators=n_estimators)
+            model.fit(X, y)
+            pred = model.predict(X, predict_model=predict_model)
+        elif task == 'classification':
+            model = cuml.ensemble.RandomForestClassifier(
+                n_estimators=n_estimators)
+            model.fit(X, y)
+            pred = model.predict_proba(X)
+        return model, pred
+    elif learner == 'skl_rf':
+        assume(has_sklearn())
+        if task == 'regression':
+            model = sklrfr(
+                n_estimators=n_estimators)
+            model.fit(X, y)
+            pred = model.predict(X)
+        elif task == 'classification':
+            model = sklrfc(
+                n_estimators=n_estimators)
+            model.fit(X, y)
+            pred = model.predict_proba(X)
+        return model, pred
+    elif learner == 'lgbm':
+        assume(has_lightgbm())
+        if task == 'regression':
+            model = lgb.LGBMRegressor(
+                n_estimators=n_estimators).fit(X, y)
+        elif task == 'classification':
+            model = lgb.LGBMClassifier(
+                n_estimators=n_estimators).fit(X, y)
+        return model.booster_, model.predict(X, raw_score=True)
+
+
+@st.composite
+def shap_strategy(draw):
+    task = draw(st.sampled_from(['regression', 'classification']))
+
+    n_estimators = draw(st.integers(1, 16))
+    n_samples = draw(st.integers(2, 100))
+    n_features = draw(st.integers(2, 100))
+    learner = draw(st.sampled_from(['xgb', 'rf', 'skl_rf', 'lgbm']))
+    supports_categorical = learner in ['xgb', 'lgbm']
+    supports_nan = learner in ['xgb', 'lgbm']
+    if task == 'classification':
+        n_targets = draw(st.integers(2, 5))
+    else:
+        n_targets = 1
```

Comment:
So `n_targets` means `n_classes` in the context of classification? Let's just use `n_classes` for this purpose, since it's confusing otherwise. (I was wondering if we were using an unreleased feature of XGBoost)

Reply:
These tests will support multi-output regression. I am using `n_targets' as a more generic term.